File: java/009 Palindrome Number/PalindromeNumber.java

/*
* 本题题意：判斷回文數字
*/
public class PalindromeNumber{
  public boolean isPalindrome(int x) {
		String str = x + "";
		int length = str.length();
		for (int i = 0, n = length / 2; i < n; i++) {
			if (str.charAt(i) != str.charAt(length - 1 - i)) {
				return false;
			}
		}
		return true;
	}
}

File: kt/009 Palindrome Number/PalindromeNumber.kt

/*
* 本题题意：判斷回文數字
*/
class PalindromeNumber {
	fun isPalindrome(x: Int): Boolean {
		val str = x.toString() + ""
		val length = str.length
		var i = 0
		val n = length / 2
		while (i < n) {
			if (str[i] != str[length - 1 - i]) {
				return false
			}
			i++
		}
		return true
	}
}
